src/Api/Endpoints/Candidate/Visit/Image/Qc.php

```diff
@@ -0,0 +1,207 @@
+        default:
+            return new \LORIS\Http\Response\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+
+        }
+    }
+
+    /**
+     * Create an array representation of this endpoint's reponse body
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request): ResponseInterface
+    {
+        if (!isset($this->_cache)) {
+            $view = (new \LORIS\Api\Views\Visit\Image\Qc(
+                $this->_visit,
+                $this->_image
+            ))->toArray();
+
+            $this->_cache = new \LORIS\Http\Response\JsonResponse($view);
+        }
+        return $this->_cache;
+    }
+
+    /**
+     * Handles a PUT request that replace an image qc status
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePUT(ServerRequestInterface $request): ResponseInterface
+    {
+        $user = $request->getAttribute('user');
+        $data = json_decode((string) $request->getBody(), true);
+
+        $inputcandid = $data['Meta']['CandID'] ?? null;
+        if ($inputcandid != $this->_visit->getCandID()) {
+            return new \LORIS\Http\Response\BadRequest(
+                'Candidate from URL does not match JSON metadata.'
+            );
+        }
+
+        $inputvisit = $data['Meta']['Visit'] ?? null;
+        if ($inputvisit != $this->_visit->getVisitLabel()) {
+            return new \LORIS\Http\Response\BadRequest(
+                'Visit from URL does not match JSON metadata.'
+            );
+        }
+
+        $inputfilename = $data['Meta']['File'] ?? null;
+        if ($inputfilename != $this->_image->getFileInfo()->getFilename()) {
+            return new \LORIS\Http\Response\BasRequest(
+                'File name from URL does not match JSON metadata.'
+            );
+        }
+
+        $inputqcstatus = $data['QCStatus'] ?? null;
+        $inputselected = $data['Selected'] ?? null;
+
+        // TODO :: This is (and was) not checking or handling Caveats
+        try {
+            $this->_image->saveQcStatus(
+                new \LORIS\ImageQcStatus(
+                    $inputqcstatus,
+                    $inputselected
+                )
+            );
+        } catch (\DatabaseException $e) {
+            error_log($e->getMessage());
+            return new \LORIS\Http\Response\InternalServerError(
+                'QC status not updated.'
+            );
+        } catch (\DomainException $e) {
+            return new \LORIS\Http\Response\BadRequest(
+                $e->getMessage()
+            );
+        } catch (\Throwable $e) {
+            return  new \LORIS\Http\Response\BadRequest();
+        }
+
+        return (new \LORIS\Http\Response())
+            ->withStatus(204);
```

Comment:
Is it worth making a custom LORIS response for this? Not in this PR but as an issue?

Reply:
Yes, 204 should have an empty body